src/utils/logResponse.ts

```diff
@@ -0,0 +1,21 @@
+import fs from 'fs';
+
+import { RateSourceResponse } from '../services/rateSource/RateSourceService';
+
+interface Input {
+  source: string;
+  response: RateSourceResponse;
+}
+
+const formatLogData = (input: Input): string =>
+  `${+new Date()} source: ${input.source} - ${input.response.code} - rate: ${
+    input.response.rate ?? 'n/a'
+  } - error: ${input.response.errorMessage ?? 'n/a'}\n`;
+
+const logResponse = (input: Input): void => {
+  fs.appendFile('logs.txt', formatLogData(input), function (err) {
+    if (err) throw err;
```

Comment:
I think logget isn't the optimal place for throwing errors since throwing an exception changes the flow of your program and will lead to crash if not handled

What about just logging them with prefix [ERR] or something and only throwing from code?